Source/santad/DataLayer/WatchItems.mm

```diff
@@ -96,34 +116,99 @@ bool ConfirmValidWatchItemConfig(const NSDictionary *watch_item_dict) {
     }
   }];
 
+  // Check the allowed cdhashes contain valid hex encoded data
+  if (success) {
+    [watch_item_dict[kWatchItemConfigKeyAllowedCDHashes]
+      enumerateObjectsUsingBlock:^(NSString *obj, NSUInteger idx, BOOL *stop) {
+        success = ConfirmValidHexString(obj, CS_CDHASH_LEN * 2);
+        if (!success) {
+          *stop = YES;
+        }
+      }];
+  }
+
+  // Check the allowed certificate hashes contain valid hex encoded data
+  if (success) {
+    [watch_item_dict[kWatchItemConfigKeyAllowedCertificatesSha256]
+      enumerateObjectsUsingBlock:^(NSString *obj, NSUInteger idx, BOOL *stop) {
+        success = ConfirmValidHexString(obj, CC_SHA256_DIGEST_LENGTH * 2);
+        if (!success) {
+          *stop = YES;
+        }
+      }];
+  }
+
   return success;
 }
 
-static std::set<std::string> ConfigArrayToSet(NSArray<NSString *> *array) {
+static std::set<std::string> StringArrayToSet(NSArray<NSString *> *array) {
   std::set<std::string> strings;
 
   for (NSString *obj in array) {
     strings.insert(std::string([obj UTF8String]));
   }
+
   return strings;
 }
 
+static inline uint8_t HexCharToByte(int c) {
+  if (c > '9') {
```

Comment:
Sure. Not a super good reason, it just seemed to handle a lot more situations than I cared for since I already checked it was a legitimate hex string. But fine to change if we think it's more readable.